server/game/cards/Mono-Expansions/IceTrap.js

```diff
@@ -8,7 +8,8 @@ class IceTrap extends Card {
                 onCardEntersPlay: (event) =>
                     BattlefieldTypes.includes(event.card.type) &&
                     event.card.life <= 2 &&
-                    !event.card.anyEffect('concealed')
+                    !event.card.anyEffect('cannotBeSpellTarget') &&
```

Comment:
it'd be nice if we didn't look at abilities here and instead just used the centralized canTarget function somehow. so that this logic is not repeated across cards. for example i just realized im able to use rin's fury on a squall stallion, which i don't think i should be able to do